### PR TITLE
Add SyTest Support

### PR DESCRIPTION
**Pull request by valkum**

### CODE REVIEW

File: src/database.rs [hunk 1]
```diff
@@ -28,16 +30,22 @@ impl Database {
...
+    pub fn load_or_create(config: &Config) -> Self {
+        let hostname = config.get_str("hostname").unwrap_or("localhost");
+        let server_name = config.get_str("server_name").unwrap_or("localhost");
+        let mut fallback_path = ProjectDirs::from("xyz", "koesters", "conduit")
             .unwrap()
             .data_dir()
             .to_path_buf();
-        path.push(hostname);
+            fallback_path.push(server_name);
+        let path = config.get_str("database_path").unwrap_or(&fallback_path.to_str().unwrap());
```
Comment (timokoesters) [inline src/database.rs]: Can you use unwrap_or_else here and calculate the fallback path in there?

File: src/main.rs [hunk 2]
```diff
@@ -18,8 +18,25 @@ pub use pdu::PduEvent;
...
+    // short and long flags (-d, --debug) will be deduced from the field's name
+    #[structopt(long, short)]
+    config: Option<String>,
+}
+
+
 
 fn setup_rocket() -> rocket::Rocket {
+
+
```
Comment (timokoesters) [inline src/main.rs]: Run cargo fmt

File: src/main.rs [hunk 3]
```diff
@@ -18,8 +18,25 @@ pub use pdu::PduEvent;
...
+#[derive(Debug, StructOpt)]
+#[structopt(name = "conduit", about = "Matrix Homeserver written in rust using rocket and ruma.")]
+pub struct CmdOptions {
+    /// Activate debug mode
+    // short and long flags (-d, --debug) will be deduced from the field's name
+    #[structopt(long, short)]
+    config: Option<String>,
+}
+
+
```
Comment (timokoesters) [inline src/main.rs]: I don't think we need the structopt dependency just for this. Can you remove this and just use std::env::args directly?

**Review by timokoesters — CHANGES_REQUESTED** [review]

**Comment by valkum** [comment]
Got merged upstream

**Pull request closed without merging**